linode_metadata/metadata_client.py

```diff
@@ -8,44 +8,55 @@
 import base64
```

Comment:
It might make sense to break these classes out into separate files given how large this file is getting

Reply:
Thanks, I think that's a good idea! Can we do the refactor in a following up PR as this PR focus on the watchers?